### ISSUE TITLE
Recursion Error when running desman

### ISSUE THREAD

**Issue by vnsriniv** [chrisquince/DESMAN#24]
Hi 

I get the following error when running the desman script:

 loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3511, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3359, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3376, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3964, in parseImpl
    loc, tokens = self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3511, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3698, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3359, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 3430, in parseImpl
    loc2 = e.tryParse( instring, loc )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1421, in tryParse
    return self._parse( instring, loc, doActions=False )[0]
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 1383, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 2779, in parseImpl
    ret = ParseResults(result.group())
  File "/home/vnsriniv/miniconda3/lib/python3.6/site-packages/pkg_resources/_vendor/pyparsing.py", line 365, in __init__
    self.__tokdict = dict()
RecursionError: maximum recursion depth exceeded while calling a Python object

My code is as follows:
```
for g in 2 3 4 5 6 7 9; do     
    for r in 0 1 2 3 4; do             
        desman ../variants/outputsel_var.csv -e ../variants/outputtran_df.csv -o cluster_RC_MAG_00014_${g}_${r} -r 1000 -i 100 -g $g -s $r > cluster_RC_MAG_00014_${g}_${r}.out&                 
    done;
done
```

What could be the issue here? Any help would be much appreciated!

**Comment by chrisquince** [comment]
Hi,
I really have no idea on this. We do have a new release now, perhaps try that? What happens when you just type:

desman

Thanks,
Chris

**Comment by vnsriniv** [comment]
Hi Chris,

Thanks for your reply. The version I am using is the current release (python 3 compatible). The same error happens when I type in just desman. Sounds like it might be a package issue or something more fundamental than with desman itself. I will try to reinstall inside a virtual environment to see if the issue resolves itself. 

Thanks
Varun

**Comment by vnsriniv** [comment]
Hi Chris,

Just an update for the record. I resolved the issue by installing a python 3.5 virtual environment and then installing desman again. This seemed to resolve the issue. Something with my python 3.6 install must have been messed up.. I am closing the issue! 

Varun
